pkg/manager/member/config.go

```diff
@@ -71,15 +80,31 @@ func updateConfigMapIfNeed(
 			return perrors.AddStack(err)
 		}
 
-		err = updateConfigMap(existing, desired)
+		dataEqual, err := updateConfigMap(existing, desired)
 		if err != nil {
 			return err
 		}
 
 		AddConfigMapDigestSuffix(desired)
+
+		confirmNameByData(existing, desired, dataEqual)
+
 		return nil
 	default:
 		return perrors.Errorf("unknown config update strategy: %v", configUpdateStrategy)
 
 	}
 }
+
+// confirmNameByData is used to fix the problem that
+// when configUpdateStrategy is changed from InPlace to RollingUpdate for the first time,
+// the name of desired configmap maybe different from the existing one while
+// the data of them are the same, which will cause rolling update.
+func confirmNameByData(existing, desired *corev1.ConfigMap, dataEqual bool) {
+	if dataEqual && existing.Name != desired.Name {
+		desired.Name = existing.Name
+	}
+	if !dataEqual && existing.Name == desired.Name {
+		desired.Name = fmt.Sprintf("%s-new", desired.Name)
```

Comment:
This will create a new CM if `InPlace` is set?
What's the issue to fix with this block?

Reply:
If `InPlace` is set, it will just update current CM, the code is [here](https://github.com/pingcap/tidb-operator/blob/master/pkg/manager/member/config.go#L58-L62).